File: 338-counting-bits/338-counting-bits.cpp
class Solution {
public:
    vector<int> countBits(int n) {
        vector<int> ans;
        ans.push_back(0);
        for(int i=1;i<=n;i++){
            int t=0;
            int p=i;
            while(p!=0){
                if(p&1!=0)t++;
                p=p>>1;
            }
            ans.push_back(t);
        }
        return ans;
        
    }
};

File: 338-counting-bits/338-counting-bits.java
class Solution {
    public int[] countBits(int n) {
        int count;
        int j=0;
        int arr[]=new int[n+1];
       for(int i=0;i<=n;i++)
       {   count=0;
           
            int x=i;
           while(x!=0)
           {
             if((x&1)==1) 
                 count++;
               x=x>>1;
           }
        arr[j++]=count;
       }
        return arr;
    }
}